### ISSUE TITLE
Azure Table Query Bug

### ISSUE THREAD

**Issue by boredom2** [Azure/azure-storage-php#22]
I observe extremely strange Query Results when querying a Table Storage Database - I may have an error in my code, but I cannot understand the results at all.

Very Simple Approach:

```
$connectionString="DefaultEndpointsProtocol=https;AccountName=...";
$tableRestProxy = \MicrosoftAzure\Storage\Common\ServicesBuilder::getInstance()->createTableService($connectionString);

$filter="PartitonKey eq 'shutdown'";
$result=$tableRestProxy->queryEntities(TABLE,$filter);
```

RESULT IS EMPTY (but shouldnt, there are tons of items with this PartitonKey)

Now I try to use a correct QueryEntitiesOptions Object:

```
$filter=new \MicrosoftAzure\Storage\Table\Models\QueryEntitiesOptions();
$filter->setFilter("PartitionKey eq 'shutdown'");
$result=$tableRestProxy->queryEntities(TABLE,$filter);
```

RESULT IS EMPTY (but shouldnt, there are tons of items with this PartitonKey)

Now I add a Projection:

```
$filter=new \MicrosoftAzure\Storage\Table\Models\QueryEntitiesOptions();
$filter->setFilter("PartitionKey eq 'shutdown'");
$filter->addSelectField('PartitionKey');
$result=$tableRestProxy->queryEntities(TABLE,$filter);
```

RESULT IS NOT EMPTY (many items are returned, but from all possible PartitionKeys, not only the one specified in the Filter).

So basically, Im completely without any clue here :)
Its just a normal Table (ok, some fields contain really huge Text Strings, thats why I added the Projection) - but none of these responses is correct.

I can send you the Endpoint Data privately, if you wanna try. It would be really great, if thats my fault, because we need this Logic on a big scale very soon.

Thanks,
Christoph

**Comment by boredom2** [comment]
Just as a sidenode, when I enter the same Query in the "Microsoft Azure Storage Explorer", I receive the correct Entries. So something is going wrong here (or I make a big mistake).

Thanks!

**Comment by boredom2** [comment]
In the meantime, I see the exactly same Behaviour in all Queries. Whatever Filter I define, the results are wrong (or there are no results at all). Entering the exact same Query in any other Tool - I get correct Results.
So - on the one hand, I always make an Error with my Queries/Filters.
Or - the Query is somehow given in a wrong way to the API.

**Comment by nowenL** [comment]
QueryEntitiesOptions.setFilter requires an Filter object instead of a string as parameter. You can change your code to:
```php
$filter = new \MicrosoftAzure\Storage\Table\Models\QueryEntitiesOptions();
$filter->setFilter(Filter::applyQueryString("PartitionKey eq 'shutdown'"));
$result = $tableRestProxy->queryEntities(TABLE, $filter);
```
or you can pass a string directly to queryEntities.

```php
$filter = "PartitionKey eq 'shutdown'";
$result = $tableRestProxy->queryEntities(TABLE, $filter);
```

**Comment by boredom2** [comment]
You are absolutelty right, my fault.
The second example is still not working propperly for whatever reason, but I agree, with QueryEntitiesOptions and applyQueryString, it works.
Close that Issue, my Fault.
